modules/components/pages/inputs/pg_stream.adoc

```diff
@@ -0,0 +1,414 @@
+- https://neon.tech/docs/guides/logical-replication-guide[Neon^]
+
+=== Replication process
+
+When you run a pipeline that uses the `pg_stream` input, Redpanda Connect makes a connection to your PostgreSQL database and creates a replication slot. The replication slot uses a copy of the Write-Ahead Log (WAL) file to subscribe to changes in your database records as they are applied to the database itself.
+
+Once the replication slot is available, Redpanda Connect:
+
+. Takes a snapshot of the current database.
+. Starts to stream data updates. It processes messages from the latest offset.
+
+[NOTE]
+====
+
+You can configure this input to run in one of two modes, to either:
+
+- Stream all data in the snapshot and data updates after the snapshot is taken
+- Stream just the data updates
+
+For full details, see the <<stream-snapshot,`stream_snapshot` field>>.
+====
+
+=== Monitoring the replication process
+
+You can monitor the initial replication of data using the following metrics:
+
+- `replication_lag_bytes`: Indicates how far the connector is lagging behind the source database when processing the transaction log.
+- `postgres_snapshot_progress`: Shows the progress of snapshot processing for each table.
+
+Once the initial replication process is complete, the snapshot is removed and the input keeps a connection open to the database so that it can receive data updates.
+
+=== Troubleshooting replication failures
+
+If the database snapshot fails, the replication slot has only an incomplete record of the existing data in your database. To maintain data integrity, you must drop the replication slot manually in your source database and run the Redpanda Connect pipeline again.
+
+```SQL
+SELECT pg_drop_replication_slot(SLOT_NAME);
+```
+
+== Metadata
+
+This input adds the following metadata fields to each message:
+
+- `mode`: Whether a message is part of the snapshot processing of existing data in a database (`snapshot`) or a data update streamed to Redpanda Connect (`streaming`).
+- `table`: The name of the database table from which the message originated.
+- `operation`: The type of database operation that generated the message, such as `INSERT`, `UPDATE` or `DELETE`.
+
+== Fields
+
+=== `dsn`
+
+The data source name (DSN) of the PostgreSQL database from which you want to stream updates. Use the format `postgres://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]`. For example, if you wanted to disable SSL in a secure environment, you would add `sslmode=disable` to the connection string.
+
+*Type*: `string`
+
+```yml
+# Examples
+
+dsn: postgres://foouser:foopass@localhost:5432/foodb
+
+dsn: postgres://foouser:foopass@localhost:5432/foodb?sslmode=disable
+```
+
+=== `batch_transactions`
+
+When set to `true`, all data updates in a transaction are batched into a single message.
+
+include::components:partial$secret_warning.adoc[]
+
+*Type*: `bool`
+
+*Default*: `true`
```

Comment:
This is gone now.

Instead you can optionally emit BEGIN and COMMIT statements: https://github.com/redpanda-data/connect/blob/main/docs/modules/components/pages/inputs/pg_stream.adoc#include_transaction_markers

If you want to create a message batch per transaction you can do that with a custom [batch policy](https://docs.redpanda.com/redpanda-connect/configuration/batching/#batch-policy) that uses `check` to look for `COMMIT` messages, then one can drop the markers using postprocessing (as mentioned in the link).

Every message is always a single change from postgres now.

Reply:
Added in the field and changed updated the metadata field descriptions.